packages/next/src/lib/helpers/get-registry.ts

```diff
@@ -8,22 +8,38 @@ import { getFormattedNodeOptionsWithoutInspect } from '../../server/lib/utils'
  * @default https://registry.npmjs.org/
  */
 export function getRegistry(baseDir: string = process.cwd()) {
+  const pkgManager = getPkgManager(baseDir)
+  // Since `npm config` command fails in npm workspace to prevent workspace config conflicts,
+  // add `--no-workspaces` flag to run under the context of the root project only.
+  // Safe for non-workspace projects as it's equivalent to default `--workspaces=false`.
+  // x-ref: https://github.com/vercel/next.js/issues/47121#issuecomment-1499044345
+  // x-ref: https://github.com/npm/statusboard/issues/371#issue-920669998
```

Comment:
New implementation has way more confidence given the explainer. This is valueable context, thank you!